lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -2785,6 +2786,28 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
                 .highlight(OA->getLocation());
           }
           OA->setInvalid();
+        } else if (auto outerTy = DC->getDeclaredInterfaceType()) {
+          // This is an overridden subscript; if the outer type has
+          // `@dynamicMemberLookup` but isn't directly annotated with it (e.g.,
+          // it's inherited), its members won't go through
+          // `@dynamicMemberLookup` checking via `AttributeChecker`. We need to
+          // ensure that if the overridden decl was a valid dynamic member
+          // lookup subscript, this override remains valid.
+          if (outerTy->hasDynamicMemberLookupAttribute()) {
+            auto D = outerTy->getAnyNominal();
+            if (!D->getAttrs().hasAttribute<DynamicMemberLookupAttr>()) {
+              if (overriddenDecl
+                      ->getDynamicMemberLookupSubscriptEligibility() !=
+                  DynamicMemberLookupSubscriptEligibility::None) {
+                auto accessScope = D->getFormalAccessScope();
+                if (SD->evaluateDynamicMemberLookupEligibility(
+                        &accessScope, &SD->getDiags()) ==
+                    DynamicMemberLookupSubscriptEligibility::None) {
+                  SD->setInvalid();
```

Comment:
Let's not add new calls to `setInvalid()`, it's kind of a bogus operation

Reply:
I think this is too late for this sort of check, it should be part of override checking.